packages/testing/src/chai-retry-plugin/chai-retry-plugin.ts

```diff
@@ -0,0 +1,109 @@
+
+const { Assertion } = Chai;
+
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace Chai {
+        interface Assertion {
+            /**
+             * Allows to retry the function passed to `expect` and assert the result until retries ended or timeout exceeded
+             * @param options Settings for retry logic:
+             * - `timeout`: The maximum duration in milliseconds to wait before failing the retry operation.
+             * - `retries`: The number of times to retry the function before failing.
+             * - `delay`: The delay in milliseconds between retries.
+             * @default { timeout: 5000, delay: 0, retries: Infinity }
+             */
+            retry(options?: RetryOptions): PromiseLikeAssertion;
+        }
+    }
+}
+
+/**
+ * Adds the `retry` method to Chai assertions, which allows to check the return value of a function until it satisfies the chained assertions.
+ * Should be applied through `Chai.use` function, for example:
+ * @example
+ * ```ts
+ * import Chai from 'chai';
+ * import { chaiRetryPlugin } from '@wixc3/testing';
+ *
+ * Chai.use(chaiRetryPlugin);
+ * ```
+ *
+ * Examples of usage:
+ * @example
+ * ```ts
+ * await expect(funcToRetry).retry({ timeout: 7000, retries: 5 }).have.property('value').and.be.above(4);
+ * ```
+ * @example
+ * ```ts
+ * await expect(sometimesNullFunction).retry({ timeout: 2000 }).to.be.not.null;
+ * ```
+ * @example
+ * ```ts
+ * await expect(funcToRetry).retry({ retries: 5, delay: 10 }).and.have.property('success').and.be.true;
+ * ```
+ */
+export const chaiRetryPlugin = function (_: typeof Chai, utils: Chai.ChaiUtils) {
+    Assertion.addMethod('retry', function (retryOptions: RetryOptions = {}): PromiseLikeAssertion {
+        const functionToRetry: FunctionToRetry = this._obj as FunctionToRetry;
+
+        if (typeof functionToRetry !== 'function') {
+            throw new TypeError(utils.inspect(functionToRetry) + ' is not a function.');
+        }
+
+        const assertionStack: AssertionStackItem[] = [];
+        const defaultRetryOptions: Required<RetryOptions> = { timeout: 5000, retries: Infinity, delay: 0 };
+        const options: Required<RetryOptions> = { ...defaultRetryOptions, ...retryOptions };
+
+        // Fake assertion object for catching calls of chained methods
+        const proxyTarget = new Assertion({});
+
+        const assertionProxy: PromiseLikeAssertion = Object.assign(
+            new Proxy(proxyTarget, {
+                get: function (target: Chai.Assertion, key: string, proxySelf: Chai.Assertion) {
+                    let value: Chai.Assertion | undefined = undefined;
+                    try {
+                        value = target[key as keyof Chai.Assertion] as Chai.Assertion;
+                    } catch {
+                        // to catch AssertionError of getter properties that immediately perform the assertion
```

Comment:
The main point of this comment is to tell why the `target[key]` might throw, so let's it (with minor alterations) above the`value = target[`. line

Reply:
@yurii-ve fixed